### PR TITLE
fix: Making ninja output on windows to work

### PR DESCRIPTION
**Pull request by refack**

### CODE REVIEW

**Comment by bnoordhuis** [comment]
I'm a little confused which PR I should review, this one or #1120?

Also, cc @kunalspathak.

File: 0005-ninja-make-ninja-work-on-windows-again.patch [hunk 1]
```diff
@@ -0,0 +1,38 @@
+From 06cea25b3af096b7fe82c25cc06ecba808ba9765 Mon Sep 17 00:00:00 2001
```
Comment (kunalspathak) [inline 0005-ninja-make-ninja-work-on-windows-again.patch]: We don't need this file i believe?

**Comment by refack** [comment]
@bnoordhuis This one is more complete. But requires a change in `GYP`, so I understand any pushback.
#1120 should be a no brainer ;)

File: 0005-ninja-make-ninja-work-on-windows-again.patch (hunk 1 above)
Comment (refack) [inline 0005-ninja-make-ninja-work-on-windows-again.patch]: We're not doing that anymore?

Comment (kunalspathak) [inline 0005-ninja-make-ninja-work-on-windows-again.patch]: never mind, i thought this was added accidentally.

File: addon.gypi [hunk 2]
```diff
@@ -109,7 +109,7 @@
           '-luuid.lib',
           '-lodbc32.lib',
           '-lDelayImp.lib',
-          '-l"<(node_root_dir)/$(ConfigurationName)/<(node_lib_file)"'
+          '-l<(node_root_dir)/<(CONFIGURATION_NAME)/<(node_lib_file)'
```
Comment (kunalspathak) [inline addon.gypi]: won't this stop working for `MSBuild` ? Where is `(CONFIGURATION_NAME) defined ?

Comment (refack) [inline addon.gypi]: <(XXXXX) is `GYP` variable resolution (i.e. `GYP` will resolve this)

Comment (kunalspathak) [inline addon.gypi]: Sorry, may be i was not clear. What i mean is for `MsBuild`, this variable is declared [here](https://github.com/nodejs/node-gyp/blob/master/gyp/pylib/gyp/MSVSSettings.py#L426) which makes it available here as well as at other places. I am wondering, with this change if you build using `MsBuild`, what value will this variable resolve to?

Comment (refack) [inline addon.gypi]: The process requires that `GYP(CONFIGURATION_NAME) === MsBuild(ConfigurationName)`
So now this resolution is a serpentius way, `GYP` sets it for `MsBuild`, and here `MsBuild` resolves it. And obviously it breaks for `ninja`/`win32`.

Comment (refack) [inline addon.gypi]: Also #1118

**Comment by rvagg** [comment]
closing due to age, hopefully this is taken up in refack/GYP and we can pull it in there?

**Pull request closed without merging**